Lib/test/test_dynamic.py

```diff
@@ -149,5 +149,49 @@ def __missing__(self, key):
         for _ in range(30):
             self.assertEqual(sum_func(), expected)
 
+class TestTracing(unittest.TestCase):
+
+    def setUp(self):
+        self.addCleanup(sys.settrace, sys.gettrace())
+        sys.settrace(None)
+
+    def test_after_specialization(self):
+
+        def trace(frame, event, arg):
+            return trace
+
+        turn_on_trace = False
+
+        class C:
+            def __init__(self, x):
+                self.x = x
+            def __del__(self):
+                if turn_on_trace:
+                    sys.settrace(trace)
+
+        def f():
+            # LOAD_GLOBAL[_BUILTIN] immediately follows the call to C.__del__
+            C(0).x, len
+
+        def g():
+            # BINARY_SUSCR[_LIST_INT] immediately follows the call to C.__del__
+            [0][C(0).x]
+
+        def h():
+            # BINARY_OP[_ADD_INT] immediately follows the call to C.__del__
+            0 + C(0).x
+
+        for func in (f, g, h):
+            with self.subTest(func.__name__):
+                for _ in range(58):
```

Comment:
I suppose 58 is enough to make the function "hot". But what if in the future the hotness limit is increased? It seems the test would then just pass regardless. Ideally the test should catch that.

Reply:
We should unify these as a constant in `test.support`. We have another constant here in `test_dis` that we have to update if we ever change the value https://github.com/python/cpython/blob/main/Lib/test/test_dis.py#L761.